lib/puppet/indirector/catalog/rest.rb

```diff
@@ -36,9 +36,6 @@ def find(request)
       static_catalog: request.options[:static_catalog],
       checksum_type: checksum_type
     )
-    # current tests rely on crazy behavior introduced in 089ac3e37dd
-    catalog.name = request.key
```

Comment:
This seems fine for catalogs, but need to check on other indirected types that may not have a "name" 
 like `file_content`

Reply:
I think this is ok